### ISSUE TITLE
All zero string decimals (e.g., '100.00') strips the '.00' and displays as '100.'

### ISSUE THREAD

**Issue by codypearce** [s-yadav/react-number-format#533]
#### Describe the issue and the actual behavior
Values in the app I am working on are passed like this '100.00', with a `decimalScale` of 2, and `isNumericString`, like this.

```jsx
<NumberFormat
        name={"AmountCents"}
        value={"100.00"}
        thousandSeparator
        prefix="$"
        decimalScale={2}
        isNumericString
      />
```

The result in the app 
<img width="207" alt="Screen Shot 2021-05-07 at 4 39 22 AM" src="https://user-images.githubusercontent.com/12564956/117444350-36d04380-aeee-11eb-8d26-82b71512b073.png">

#### Describe the expected behavior
Since it was passed in as `100.00` it should display as `100.00`. 

The changes in #501 appear to be the issue. 
```js
var floatValue = parseFloat("0.".concat(afterDecimal || '0'));
var floatValueStr = afterDecimal.length <= scale ? floatValue.toString() : floatValue.toFixed(scale);
```
`floatValue` is `0`
`afterDecimal.length` is 2 and `scale` is 2
so `floatValueStr` returns `floatValue.toString()` which returns `0`
However, `floatValueStr` should probably return `0.00`

#### Provide a CodeSandbox link illustrating the issue
https://codesandbox.io/s/vigilant-shamir-dk4u9?file=/src/App.js

#### Provide steps to reproduce this issue
See codesandbox

#### Please check the browsers where the issue is seen
- [x] Chrome
- [ ] Chrome (Android)
- [x] Safari (OSX)
- [ ] Safari (iOS)
- [ ] Firefox
- [ ] Firefox (Android)

**Comment by nikhil-varma** [comment]
Hey @codypearce Thanks for reporting this issue. We will investigate and shoot out a fix soon. Keep an eye out for updates 🙂

**Comment by nikhil-varma** [comment]
@codypearce Just a quick work around until the issue is being investigated. You can use the `fixedDecimalScale` prop to match the leading zeros.  Here's a sample Codesandbox link: https://codesandbox.io/s/zealous-oskar-wjnqh

From the document for `fixedDecimalScale` 
> If true it add 0s to match given decimalScale

I will get back to you with an update on the issue resolution. Thank you for your patience.

**Comment by nikhil-varma** [comment]
@codypearce Update on the issue

So the problem is that all 0s after the decimal point becomes an exceptional case for conditions `roundedDecimalParts[1] || ''`. So instead of passing, -in this case, `00` to the `limitToScale` function, it passes `''` and that created this issue. 

The PR #501 solved this issue #495 . Please refer to that for the idea of the resolution.

The PR is in review and will get merged shortly. Appreciate your patience and time for reporting this issue! Thanks!

**Comment by nikhil-varma** [comment]
@codypearce This issue is now fixed and released in v4.6.2 

Closing this issue in light of the fix. Feel free to open/ask questions if required.
